src/algebra/group/hom.lean

```diff
@@ -1,18 +1,51 @@
 /-
-Copyright (c) 2014 Patrick Massot. All rights reserved.
+Copyright (c) 2018 Patrick Massot. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Patrick Massot, Kevin Buzzard, Scott Morrison, Johan Commelin, Chris Hughes, Johannes Hölzl, Yury Kudryashov
 
 Homomorphisms of multiplicative and additive (semi)groups and monoids.
+
 -/
+
 import algebra.group.to_additive algebra.group.basic
 
+/-!
+# monoid and group homomorphisms
+
+This file defines the basic structures for monoid and group
+homomorphisms, both unbundled (e.g. `is_monoid_hom f`) and bundled
+(e.g. `monoid_hom f`). The unbundled ones are deprecated and the plan is
+to slowly remove them from mathlib.
+
+## main definitions
+
+moniod_hom, is_monoid_hom (deprecated), is_group_hom (deprecated)
+
+## Notations
+
+→* for bundled monoid homs (also use for group homs)
+→+ for bundled add_monoid homs (also use for add_group homs)
+
+## implementation notes
+
+There's a coercion from bundled homs to fun, and the canonical
+notation is to use the bundled hom as a function via this coercion.
+
+There is no `group_hom` -- the idea is that `monoid_hom` is used.
```

Comment:
Maybe point out the special constructor for group homs.

Reply:
OK!